baseframe/filters.py

```diff
@@ -178,36 +179,37 @@ def cdata(text):
     return Markup('<![CDATA[' + text.replace(']]>', ']]]]><![CDATA[>') + ']]>')
 
 
-@baseframe.app_template_filter('shortdate')
-def shortdate(value):
+@baseframe.app_template_filter('date')
+def date_filter(value, fm='medium'):
     if isinstance(value, datetime):
-        tz = get_timezone()
         if value.tzinfo is None:
-            dt = UTC.localize(value).astimezone(tz)
+            dt = UTC.localize(value).astimezone(get_timezone())
         else:
-            dt = value.astimezone(tz)
-        utc_now = request_timestamp().astimezone(tz)
+            dt = value.astimezone(get_timezone())
     else:
         dt = value
-        utc_now = request_timestamp().date()
-    if dt > (
-        utc_now
-        - timedelta(days=int(current_app.config.get('SHORTDATE_THRESHOLD_DAYS', 0)))
-    ):
-        return dt.strftime('%e %b')
+    return format_date(dt, format=fm, locale=get_locale())
+
+
+@baseframe.app_template_filter('format_time')
+def time_filter(value):
+    if isinstance(value, datetime):
+        if value.tzinfo is None:
+            dt = UTC.localize(value).astimezone(get_timezone())
+        else:
+            dt = value.astimezone(get_timezone())
     else:
-        # The string replace hack is to deal with inconsistencies in the underlying
-        # implementation of strftime. See https://bugs.python.org/issue8304
-        return six.text_type(dt.strftime("%e %b '%y")).replace(u"'", u"’")
+        dt = value
+    return format_time(dt, "hh:mm a", locale=get_locale())
```

Comment:
`format_time` also takes `format` as a preset string. Pass it in. Don't hardcode a rendering.